### PR TITLE
[docs] Move away from rnplay to snack, with embedded examples!

### PR DESCRIPTION
**Pull request by brentvatne**
## Motivation

React Native Playground has been sunset, so I've replaced the examples that previously used it with examples using [Snack](http://snack.expo.io/).

The examples are directly embedded and can be edited live to see updates. The code itself is also in the docs, so we can easily update the docs in one place and we don't have to actually go to a saved app on Snack and update it there.

### Desktop docs

Run it locally, go to the `Animations` section and the `Direct Manipulation` section.

screen shot 2017-04-03 at 6 29 51 pm

### Mobile docs

Open it on your phone, notice that it falls back to just showing plain code.

## Next Steps

- Get rid of the Expo new user experience dialog that you see when you open a Snack -- is this a dealbreaker for merging? (@terribleben and @jesseruder will work on this for iOS and Android respectively)
- Tone down the logo in the top right of the embed. We're thinking grayscale, slightly smaller, and no "Made with" text. (I'll work on this soon)
- The area where it says "Tap to Play" is loaded in an iframe and it slows down load time because there's a lot of JS being loaded in each iframe. I want to make loading faster by not loading the simulator iframe every time. We could instead load our own custom placeholder that's just a div with some text that you can click.
- Move logic around mapping react-native versions to Expo SDK versions to Snack, so we can just pass in whatever the docs version is without any fuss. (@jesseruder will work on this).
- Think of another way to set the name without having to `encodeURIComponent` on it (if you use spaces, markdown doesn't parse it as you would hope, codeblock formatting breaks).
- Remove old UIExplorer examples from docs.
- And of course, add embedded examples to more parts of the docs! Anybody can do this with relative ease.

## Longer Term Next Steps
- Refactor UIExplorer such that we are able to automatically pull examples from it into SnackPlayer
- It would be nice, eventually, to unify web-player and Snack player so we have a consistent experience across the docs. cc @dabbott

### CODE REVIEW

File: website/core/SnackPlayer.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    window.ExpoSnack && window.ExpoSnack.initialize();
+  },
+
+  render() {
+    var code = encodeURIComponent(this.props.children);
+    var params = this.parseParams(this.props.params);
+    var platform = params.platform ? params.platform : 'ios';
+    var name = params.name ? decodeURIComponent(params.name) : 'Example';
+    var description = params.description
+      ? decoreURIComponent(params.description)
```
Comment (ide) [inline website/core/SnackPlayer.js]: test this line specifically

File: website/core/SnackPlayer.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ * export default class App extends React.Component {
+ *   render() {
+ *     return <Text>Hello World!</Text>;
+ *   }
+ * }
+ * ```
+ */
+var SnackPlayer = React.createClass({
+  contextTypes: {
+    version: React.PropTypes.number.isRequried,
```
Comment (hramos) [inline website/core/SnackPlayer.js]: Typo here in `isRequired`

**Comment by hramos** [comment]
This is awesome, very excited to see an embedded Snack player in the docs! How's the performance when there's more than one Snack player on the same page?

**Comment by satya164** [comment]
Hey @hramos, not entirely sure whose performance you want to know.

The snack player is just an iframe streaming the app using appetize, so performance of simulator should not be different from the appetize emulators we use.

Having a lot of iframes can have a negative impact because iframes are blocking. But as long as we keep them to 1-2 snacks per page, it should be fine.

File: website/core/SnackPlayer.js (hunk 2 above)
Comment (brentvatne) [inline website/core/SnackPlayer.js]: derp

**Comment by hramos** [comment]
We noticed this back when we added the WebPlayer that having more than a couple on a single page could heavily impact page loading performance. 1-2 Snacks per page should be fine, I don't think we have any docs with more than a couple examples per page anyway (other than the reference docs that use the UIExplorer app).

**Comment by brentvatne** [comment]
@satya164 @hramos 

I think the main pain point is dropped frames while loading the iframe, after that afaict there is no visible impact on my MBP. I'd like to resolve this by only loading the iframe when the user requests to preview the app. The plan for this is roughly outlined in my original comment:

> The area where it says "Tap to Play" is loaded in an iframe and it slows down load time because there's a lot of JS being loaded in each iframe. I want to make loading faster by not loading the simulator iframe every time. We could instead load our own custom placeholder that's just a div with some text that you can click.

I think when we get around to this we don't have to be concerned about the number of embeds on a page. Additionally, because the embeds are fixed height we could have them only be initialized when they first become visible on screen, but I'd rather do this after the delayed iframe loading optimization because otherwise we would introduce scroll-jank as you're browsing the page rather than just initial load.

**Comment by satya164** [comment]
@brentvatne Sorry! I didn't really read what you wrote xD

**Comment by brentvatne** [comment]
Added a "Longer Term Next Steps" to the original comment, something for you to snack on (wooooooooooo boy) for now. UIExplorer automatically getting into Snack examples would be :1st_place_medal:

**Comment by hramos** [comment]
Unrelated failure. Attempting to land again.

**Pull request closed without merging**